dotCMS/src/main/java/com/dotcms/contenttype/business/SiteAndFolderResolver.java

```diff
@@ -0,0 +1,24 @@
+package com.dotcms.contenttype.business;
+
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Config;
+import com.liferay.portal.model.User;
+
+public interface SiteAndFolderResolver {
+
+    String CT_SKIP_RESOLVE_SITE = "CT_SKIP_RESOLVE_SITE";
+
+    String CT_FALLBACK_DEFAULT_SITE = "CT_FALLBACK_DEFAULT_SITE";
+
+    ContentType resolveSiteAndFolder(ContentType contentType)
+            throws DotDataException, DotSecurityException;
+
+    static SiteAndFolderResolver newInstance(final User user) {
+        return new SiteAndFolderResolverImpl(user,
+                Config.getBooleanProperty(CT_SKIP_RESOLVE_SITE, false),
```

Comment:
if this is call so often, please use a Lazy. Config is a blocking code

Reply:
done